hasura.planx.uk/migrations/1706542015549_alter_table_public_feedback_alter_column_project_type/up.sql

```diff
@@ -0,0 +1 @@
+ALTER TABLE "public"."feedback" ALTER COLUMN "project_type" TYPE text[] USING string_to_array("project_type", ',');
```

Comment:
nit: I think separate columns for `uprn`, `address`, and `project_type` are redundant if we're capturing `breadcrumbs`? Would parsing specific values out of jsonb columns like `breadcrumbs` be easier in a `view` later (that could then be read into Metabase etc) rather than repeatedly storing directly on the `feedback` table itself? Considering each of these individual fields are nullable depending on journey, they don't feel super meaningful as separate columns to me!

Reply:
That's a very good point, I suppose I was anchored on the current columns in the FeedbackFish results. Although as you say we'll have the DB view to extract the information. 

I have a suspicion that parsing the data from the breadcrumbs in the view could be challenging although I could give it a go and see how I get on.

Reply:
Ah right - parsing breadcrumbs would be tedious - perhaps the existing `breadcrumbs` column could be more general `user_data` jsonb instead with shape `{ passport: {...}, breadcrumbs: {...} }` ?

This would: 
- Better align to how we format & store `lowcal_session.data`
- Capture the full passport & breadcrumbs at point of feedback (you have the whole passport at point of updating the table anyways, and it should be more flexible for deciding which specific fields are valuable to pull out in views later) 
- Make it easier to parse "allow-list-type" values out later like `select 'passport' -> 'data' -> 'proposal.projectType'`

Reply:
I was also thinking I could drop the `component_metadata` column in favour of `breadcrumbs`/`passport` as generally this column was used to enrich feedback data:

- [constraints](https://github.com/theopensystemslab/planx-new/blob/a5deb4e76dada738797d4829c1d93d747be80f8c/editor.planx.uk/src/%40planx/components/PlanningConstraints/Public.tsx#L222-L225)
- [propertyDetails](https://github.com/theopensystemslab/planx-new/blob/a5deb4e76dada738797d4829c1d93d747be80f8c/editor.planx.uk/src/%40planx/components/PropertyInformation/Public.tsx#L119-L123)
- [responses](https://github.com/theopensystemslab/planx-new/blob/a5deb4e76dada738797d4829c1d93d747be80f8c/editor.planx.uk/src/%40planx/components/Result/Public/index.tsx#L107-L109)

Which would all be stored?

Reply:
Correct - constraints, property details, and responses can all be derived by the breadcrumbs/passport ! 

The "reason" - eg "Inaccurate property type", "Inaccurate result" - can also be inferred by the node they were on when they left feedback. I wonder if in addition to `node_id`, it'd be worth explicitly writing a `node_type` column as we know the human-readable node types are a bit easier to update from the frontend rather than join/lookup later?

Reply:
Yeah I think that's a really good idea

Reply:
Cool - thanks for adjusting data structure - I know un-doing/re-doing migrations is one of the more annoying tasks with vultr, but feels like a much more meaningful & simplified table schema now :100:

Reply:
It makes a lot of sense to get this stuff right now and a little inconvenience now I'm sure will pay dividends when we're using this table 💪 

Thanks for all the feedback and explanations 😊